doc/content/3.documentation/5.configuration/1.consumers.md

```diff
@@ -213,3 +213,42 @@ ConfigureConsumers(context);
 
 Configures all consumers that haven't been configured on the receive endpoint.
 
+## Consumer Definitions
+
+Inside of a consumer definition you can control all of the definitions about a consumer and its associated endpoint.
+
+```csharp
+public class SubmitOrderConsumerDefinition :
+    ConsumerDefinition<SubmitOrderConsumer>
+{
+    public SubmitOrderConsumerDefinition()
+    {
+        // override the default endpoint name, for whatever reason
+        EndpointName = "ha-submit-order";
+
+        // limit the number of messages consumed concurrently
+        // this applies to the consumer only, not the endpoint
+        ConcurrentMessageLimit = 4;
+    }
+
+    protected override void ConfigureConsumer(IReceiveEndpointConfigurator endpointConfigurator,
```

Comment:
this override is no longer recommended starting from 8.1.x, @drusellers 
So it is better to use this one:
https://github.com/MassTransit/MassTransit/blob/e679624edccaed406ff9bc52c05c99aeddb3a38a/src/MassTransit.Abstractions/Configuration/DependencyInjection/ConsumerDefinition.cs#L94